dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/internal/sanitizer/DdUrlSanitizer.kt

```diff
@@ -0,0 +1,41 @@
+package com.datadog.gradle.plugin.internal.sanitizer
+
+import java.util.Locale
+
+/**
+ * It will strip out the user information from the provided URL by applying the following rules:
+ * 1. In case the URL is a valid schema URL : ((ssh|http[s]?):\\) it will strip out the user
+ * information if any from it
+ * 2. In case the URL is a valid schema URL but by mistake the user forgot to add the schema:
+ * [username]:[password]@github.com/(.+).git it will strip out the user information if any and will
+ * return the new url
+ * 3. In case the URL is of a GIT SSH short format: [user@]github.com:(.+).git it will do
+ * nothing assuming that there is no password normally provided in a GIT SSH short format URL.
+ */
+internal class DdUrlSanitizer(
+    private val strategyResolver: (String) -> SanitizerStrategy = {
+        if (it.matches(Regex(VALID_SCHEMA_URL_FORMAT_REG_EX, RegexOption.IGNORE_CASE))) {
+            DefaultSanitizerStrategy()
+        } else {
+            NoSchemaSanitizerStrategy()
+        }
+    }
+) : UrlSanitizer {
+
+    @SuppressWarnings("TooGenericExceptionCaught")
+    override fun sanitize(url: String): String {
+        try {
+            return strategyResolver(url).apply(url)
+        } catch (e: Exception) {
+            throw UriParsingException(WRONG_URL_FORMAT_ERROR_MESSAGE.format(url, Locale.US), e)
```

Comment:
This is only to make sure we are not missing anything, normally this should never occur as this can only be triggered from the `URI` parsing and honestly I tried with all the combinations and as long as you have the schema set there is no way to trigger an exception from there :). We are already checking for a schema existence before going into the `URI` parser but just for the code sanity we need to make sure we are treating that possible exception. Unfortunately if there is any error in the `URI` parser the plugin will be broken as that `URL` will not be usable and we will have to fail the build.